### DISCUSSION TITLE
请问有没有静态修改所有组件class前缀的方法，ConfigProvider.config({prefixCls: 'xxxx'})不行

### DISCUSSION THREAD

**Discussion by taiyangzaitianshang** [ant-design/ant-design#40987] [Q&A] [answered]
比如想基于Antd拓展组件库
```
export { Button } from 'antd';
export { Card } from 'antd';
....

export { customComponent } from './xxx/xxx';
```
请问有没有办法让上面导出的组件在使用的时候用的是自定义前缀，可以不用再最外层自行包<ConfigProvider prefixCls='xxxx'>。

尝试使用了ConfigProvider.config({prefixCls: 'xxxx'})并不能起效果

**Comment by arvinxx** [comment]
@zombieJ  @MadCcc  能不能在ConfigProvider 加个配置来设定呢？这个真的是非常刚需。

类似 `theme.defaultConfig.token` 这样的。

**Reply by zombieJ** [reply]
`theme.defaultConfig.token` 是为了消费不是为了魔改，非 React 生命周期的修改很容易让人以为是可以外部修改触发更新的形式，实际上并不是如此。另外还有一点就是如果可以静态修改，那三方组件库就可以不引入到 jsx 中就“魔改”开发者的页面，毕竟静态方法一设就全都变了，这是非预期的。

在 v5 都推荐 hooks or App 来走生命周期。

**Reply by arvinxx** [reply]
> `theme.defaultConfig.token` 是为了消费不是为了魔改，非 React 生命周期的修改很容易让人以为是可以外部修改触发更新的形式，实际上并不是如此。另外还有一点就是如果可以静态修改，那三方组件库就可以不引入到 jsx 中就“魔改”开发者的页面，毕竟静态方法一设就全都变了，这是非预期的。
> 
> 在 v5 都推荐 hooks or App 来走生命周期。

核心不是为了外部修改触发，核心是为了改「默认值」。就像一个组件有 value ，也会有 defaultValue 一样。外部触发改 defaultValue 也就只会改第一次，后面不会改了。

现在的问题就是如果要改默认值，得每个组件都再包一层，非常麻烦…

**Comment by zombieJ** [comment, marked as answer]
没有静态修改方式，只有通过 `<ConfigProvider prefixCls="xxx" />` context 来传递。
